src/components/viewStats/components/ViewStats.js

```diff
@@ -18,56 +21,122 @@ const countGender = (oldCount, person) => {
     return newCount;
 };
 
+// reducer function used to group the candidiates by age
+const countAge = (oldCount, person) => {
+    const newCount = oldCount;
+    newCount['18-25'] += (person.age >= 18 && person.age < 26);
+    newCount['26-40'] += (person.age >= 26 && person.age < 40);
+    newCount['40-60'] += (person.age >= 40 && person.age < 60);
+    newCount['60+'] += (person.age > 60);
+    return newCount;
+};
+// function to transfor candidates from their id to their name
+const transformCandidates = candidates => {
+    const nameMap = {};
+    candidates.forEach(candidate => {
+        nameMap[candidate.id] = candidate.name;
+    });
+    return nameMap;
+};
+
 export default function ViewStats({ match }) {
     const dispatch = useDispatch();
-
+    // get the data from state
     const voters = useSelector(state => state.voters);
     const candidates = useSelector(state => state.candidates);
 
+    // create mappings to transform our data to  use for the charts
+    // get the number of each gender vote
     const genderCount = voters.reduce(countGender, [0, 0]);
+    // get the number of each age group vote
+    const voterAgeDist = voters.reduce(countAge, {
+        '18-25': 0, '26-40': 0, '40-60': 0, '60+': 0,
+    });
+    // get the name of each candidate and current votecount for piechart
     const candidatesMapped = candidates.map(candidate => ({
         name: candidate.name,
         sliced: true,
         y: Number(candidate.voteCount),
     }));
-
+    // group the voters by gender and votecount
+    const group = DataGrouper.count(voters, ['gender', 'votedCandidate']);
+    // turn the list of candidates into {candidateId:candidateVote}
+    const getCandidatesNameById = transformCandidates(candidates);
+    // get the total number of male votes for each candidate
+    const maleVote = Object.keys(getCandidatesNameById).map(id => group
+        .filter(newGroup => newGroup.gender === 'male' && newGroup.votedCandidate === id)
+        .map(newGroup => newGroup.Value)
+        .reduce((current, previous) => current + previous, 0));
```

Comment:
all of this logic can be handled in one `.reduce`. This is a very inefficient function that you just wrote